pandas/io/pytables.py

```diff
@@ -3382,6 +3382,8 @@ def read(
             if (
                 using_string_dtype()
                 and isinstance(values, np.ndarray)
+                # TODO: Should is_string_array return True for an empty object ndarray?
+                and values.size != 0
```

Comment:
This is because the Cython uses `len(values)` where I think it should be using `values.size`. Will address in a precursor.

Reply:
Opened #60796